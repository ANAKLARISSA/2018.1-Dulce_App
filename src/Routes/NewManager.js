import React from 'react';
import NewManagerNavigator from '../Navigators/NewManagerNavigator';

class NewManager extends React.Component {

  render() {
    return (
      <NewManagerNavigator />
    );
  }
}

export default NewManager;
