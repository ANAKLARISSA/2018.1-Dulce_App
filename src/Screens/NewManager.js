import React from 'react';
import {Text, View, TextInput, Image, Alert, ScrollView} from 'react-native';
import ValidationComponent from 'react-native-form-validator';
import AGRButton from '../Components/AGRButton';


const logo = require('../../assets/img/logo.png');

const styles = {
  logo: {
    alignSelf: 'center',
    marginBottom: 20,
    width: 200,
    height: 100,
    resizeMode: 'contain'
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 50,

  },
  input: {
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    width: '100%',
    alignSelf: 'center',
    marginBottom: 20,
  },
  button: {
    marginTop : 30
  },
  error:{
    color: 'red',
    alignSelf: 'center'
  }
};

export default class NewManager extends ValidationComponent {

  constructor(props) {
    super(props);
    this.state = {
      nome: '',
      matricula: '',
      hospital: '',
      setor: '',
      senha: '',
      confirmaSenha: ''
    }
  }

  _onPressButton(){
    if(this.validate({
        nome: {required: true},
        matricula: {numbers: true, required: true},
        hospital: {required: true},
        setor: {required: true},
        senha: {minlength:4, maxlength:8, required: true},
        confirmaSenha: {required: true}
      }) && (this.state.confirmaSenha == this.state.senha)){
        //Metodo para mandar para a API estará aqui
        Alert.alert(
          'Dados do novo gestor',
          'nome: ' + this.state.nome +
          '\nMatricula: ' + this.state.matricula +
          '\nHospital: ' + this.state.hospital +
          '\nSetor: ' + this.state.setor +
          '\nSenha: ' + this.state.senha +
          '\nConfirmação de Senha: ' + this.state.confirmaSenha,
          [
            {text:'Ok', onPress: () => this.props.navigation.navigate('login')}
          ]
         )
      }
  }

  confirmaSenhaErrado(){
    if (this.state.confirmaSenha != this.state.senha) {
      return true;
    }else {
      return false;
    }

  }

  render() {
    return (
      <View style={styles.container}>
        <Image source={logo} style={styles.logo} />
        <ScrollView>
          <TextInput ref="nome" style={styles.input} placeholder="Nome Completo" underlineColorAndroid="transparent" onChangeText={(text) => this.setState({nome: text})} />
          {this.isFieldInError('nome') && this.getErrorsInField('nome').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <TextInput ref="matricula" style={styles.input} placeholder="Matrícula" underlineColorAndroid="transparent" onChangeText={(text) => this.setState({matricula: text})} />
          {this.isFieldInError('matricula') && this.getErrorsInField('matricula').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <TextInput ref="hospital" style={styles.input} placeholder="Hospital" underlineColorAndroid="transparent" onChangeText={(text) => this.setState({hospital: text})} />
          {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <TextInput ref="setor" style={styles.input} placeholder="Setor" underlineColorAndroid="transparent" onChangeText={(text) => this.setState({setor: text})} />
          {this.isFieldInError('setor') && this.getErrorsInField('setor').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <TextInput ref="senha" style={styles.input} placeholder="Senha" underlineColorAndroid="transparent" secureTextEntry onChangeText={(text) => this.setState({senha: text})} />
          {this.isFieldInError('senha') && this.getErrorsInField('senha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <TextInput ref="confirmaSenha" style={styles.input}
            placeholder="Confirme sua senha"
            underlineColorAndroid="transparent"
            secureTextEntry
            onChangeText={(text) => this.setState({confirmaSenha: text})} />
          {this.isFieldInError('confirmaSenha') && this.getErrorsInField('confirmaSenha').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>)}
          <AGRButton style={styles.button} text='Criar' onPress={() => this._onPressButton()} />
        </ScrollView>
      </View>
    );
  }
}
