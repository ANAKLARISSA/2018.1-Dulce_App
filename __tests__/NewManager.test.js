import React from 'react';
import NewManager from '../src/Screens/NewManager';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

test('renders correctly', () => {
  const tree = renderer.create(<NewManager />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('change name file ', () => {
  const wrapper = shallow(<NewManager />);
  const nameField = wrapper.find('AGRInput').at(0);
  nameField.simulate('changeText','text');
  expect(wrapper.state('name')).toBe('text');
});

test('change registration file ', () => {
  const wrapper = shallow(<NewManager />);
  const registrationField = wrapper.find('AGRInput').at(1);
  registrationField.simulate('changeText','text');
  expect(wrapper.state('registration')).toBe('text');
});

test('change hospital file ', () => {

  const wrapper = shallow(<NewManager />);
  const hospitalField = wrapper.find('AGRInput').at(2);
  hospitalField.simulate('changeText','text');
  expect(wrapper.state('hospital')).toBe('text');

});

