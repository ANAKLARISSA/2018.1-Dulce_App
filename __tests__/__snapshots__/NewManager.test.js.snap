// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 15,
      "paddingHorizontal": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/img/logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 100,
        "marginBottom": 20,
        "resizeMode": "contain",
        "width": 200,
      }
    }
  />
  <RCTScrollView>
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Nome Completo"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="MatrÃ­cula"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Hospital"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Setor"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Senha"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Confirme sua senha"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "height": 36,
            "marginBottom": 20,
            "underlineColorAndroid": "transparent",
            "width": "100%",
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "grey",
            "height": 50,
            "justifyContent": "center",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Criar
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
